Yoast/duplicate-post
# Settings pages accessibility first run

This PR seeks to improve accessibility on the plugin Settings pages:

Note: needs to be rebased onto the `svntrunk` branch

- associates `<label>` elements with all form controls via for/id attributes
- groups checkboxes in `<fieldset>` elements with a visually hidden `<legend>` element
- adds `aria-describedby` attributes to associate some descriptions to their form fields
- adds aria-expanded to the button to toggle private taxonomies visibility
- also, this was a link with a `href="#"` attribute and now it's a button
- minor CSS improvements
- improves the "Settings" link on the Plugins overview page 
- git-ignores the `/vendor/` directory 
- note that my editor automatically removes trailing spaces so you will find unrelated changes to some lines, for a good reason 

Note: the 3 tabs content panels are wrapped in `<section>` elements. We may want to consider to remove the sections later, as they're not ideal for this usage (they also lack a required heading).

## To test
- clone this repo in your WordPress plugins directory and activate the plugin
- go to Settings > Duplicate Post
- check all the form controls within the 3 tabs have properly associated labels: you can do this by clicking on the labels and verifying the associated form control gets focused
- check everything works as expected
